core/src/services/hdfs/backend.rs

```diff
@@ -163,13 +194,35 @@ impl Builder for HdfsBuilder {
 pub struct HdfsBackend {
     root: String,
     client: Arc<hdrs::Client>,
+    atomic_write_dir: Option<PathBuf>,
```

Comment:
We don't need to use `PathBuf` here since all hdfs path are required to be UTF-8.